crates/beerus-core/src/lightclient/beerus.rs

```diff
@@ -471,4 +471,20 @@ impl BeerusLightClient {
             _ => Err(eyre::eyre!("Block not found")),
         }
     }
+
+    ///  Returns the pending transactions in the starknet transaction pool
+    /// See https://github.com/starknet-io/starknet-addresses for the StarkNet core contract address on different networks.
+    /// # Arguments
+    /// # Returns
+    /// `Ok(U256)` if the operation was successful - A vector of pending transactions
+    /// `Err(eyre::Report)` if the operation failed - No pending transactions found.
+    // TODO: Determine if error should throw if no pending transactions are found, or just a zero case
+    pub async fn starknet_pending_transactions(&self) -> Result<Vec<Transaction>> {
+        let transactions = self.starknet_lightclient.pending_transactions().await;
+        
+        match transactions {
```

Comment:
You cannot match here you will get a type error: 
you are only able to match on Option<> and Enum<> but here you are trying to match on a vec<Transaaction>.

You should check if the vec.len() == 0, to know if it is empty or not.

Reply:
Imo if it is empty you return the empty vector, it just means that no transactions are pending, which is possibly some informations that people want to know too.

Only return an error if you have an actual error, like a bad call or others...

Wdythink @drspacemn?

Reply:
Got it! yeah that's what I was thinking. So the error case would be if the call itself fails?